### PR TITLE
Change the code according to RestAPI

### PR DESCRIPTION
**Pull request by nick863**
# Description

During design discussion we decided to use MessageAttachment object, which will contain one data source instead of multiple data sources list. In this PR we are doing this change.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by jhakulin — APPROVED** [review]

File: sdk/ai/azure-ai-projects/samples/agents/sample_agents_code_interpreter_attachment_enterprise_search.py [hunk 1]
```diff
@@ -62,7 +60,12 @@
     ds = VectorStoreDataSource(asset_identifier=asset_uri, asset_type=VectorStoreDataSourceAssetType.URI_ASSET)
 
     # create a message with the attachment
-    attachment = MessageAttachment(data_sources=[ds], tools=code_interpreter.definitions)
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/samples/agents/sample_agents_code_interpreter_attachment_enterprise_search.py:65]: Would the earlier way also work if you have

    code_interpreter = CodeInterpreterTool()

    agent = project_client.agents.create_agent(
        model="gpt-4-1106-preview",
        name="my-assistant",
        instructions="You are helpful assistant",
        tools=code_interpreter.definitions,
    )

 attachment = MessageAttachment(data_source=ds, tools=code_interpreter.definitions)

File: sdk/ai/azure-ai-projects/samples/agents/sample_agents_code_interpreter_attachment_enterprise_search.py [hunk 2]
```diff
@@ -62,7 +60,12 @@
...
+    attachment = MessageAttachment(
+        data_source=ds,
+        tools=[
+                FileSearchTool().definitions[0],
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/samples/agents/sample_agents_code_interpreter_attachment_enterprise_search.py:66]: We should not need to define both file search and code interpreter

Comment (nick863) [reply sdk/ai/azure-ai-projects/samples/agents/sample_agents_code_interpreter_attachment_enterprise_search.py]: Fixed.

File: sdk/ai/azure-ai-projects/samples/agents/sample_agents_code_interpreter_attachment_enterprise_search.py (hunk 1 above)
Comment (nick863) [reply sdk/ai/azure-ai-projects/samples/agents/sample_agents_code_interpreter_attachment_enterprise_search.py:65]: Fixed.

**Review by howieleung — APPROVED** [review]

**Pull request merged**